library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java

```diff
@@ -175,47 +159,64 @@
             DownloadContract.Batches.COLUMN_DESCRIPTION,
             DownloadContract.Batches.COLUMN_BIG_PICTURE,
             DownloadContract.Batches.COLUMN_VISIBILITY,
-            DownloadContract.Batches.COLUMN_TOTAL_BYTES,
-            DownloadContract.Batches.COLUMN_CURRENT_BYTES,
+            DownloadContract.BatchesWithSizes.COLUMN_TOTAL_BYTES,
+            DownloadContract.BatchesWithSizes.COLUMN_CURRENT_BYTES,
             OpenableColumns.DISPLAY_NAME,
             OpenableColumns.SIZE,
     };
 
-    private static final HashSet<String> APP_READABLE_COLUMNS_SET;
-    private static final HashMap<String, String> COLUMNS_MAP;
+    private static final Set<String> APP_READABLE_COLUMNS_SET;
 
-    static {
-        APP_READABLE_COLUMNS_SET = new HashSet<>();
-        Collections.addAll(APP_READABLE_COLUMNS_SET, APP_READABLE_COLUMNS_ARRAY);
-
-        COLUMNS_MAP = new HashMap<>();
-        COLUMNS_MAP.put(OpenableColumns.DISPLAY_NAME, DownloadContract.Batches.COLUMN_TITLE + " AS " + OpenableColumns.DISPLAY_NAME);
-        COLUMNS_MAP.put(OpenableColumns.SIZE, DownloadContract.Downloads.COLUMN_TOTAL_BYTES + " AS " + OpenableColumns.SIZE);
-    }
+    private static final Map<String, String> COLUMNS_MAP;
 
     private static final List<String> DOWNLOAD_MANAGER_COLUMNS_LIST = Arrays.asList(DownloadManager.UNDERLYING_COLUMNS);
 
+    private final DownloadsUriProvider downloadsUriProvider;
+
     /**
      * Different base URIs that could be used to access an individual download
      */
     private final Uri[] baseUris;
 
+
     /**
      * The database that lies underneath this content provider
      */
-    private SQLiteOpenHelper openHelper = null;
+    private SQLiteOpenHelper openHelper;
 
     /**
      * List of uids that can access the downloads
      */
     private int systemUid = -1;
+
     private int defcontaineruid = -1;
     private File downloadsDataDir;
-
     //    @VisibleForTesting
     SystemFacade systemFacade;
 
-    private DownloadsUriProvider downloadsUriProvider;
+    static {
+        URI_MATCHER.addURI(AUTHORITY, "my_downloads", MY_DOWNLOADS);
+        URI_MATCHER.addURI(AUTHORITY, "my_downloads/#", MY_DOWNLOADS_ID);
+        URI_MATCHER.addURI(AUTHORITY, "all_downloads", ALL_DOWNLOADS);
+        URI_MATCHER.addURI(AUTHORITY, "all_downloads/#", ALL_DOWNLOADS_ID);
+        URI_MATCHER.addURI(AUTHORITY, "batches", BATCHES);
+        URI_MATCHER.addURI(AUTHORITY, "batches/#", BATCHES_ID);
+        URI_MATCHER.addURI(AUTHORITY, "downloads_by_batch", DOWNLOADS_BY_BATCH);
+        URI_MATCHER.addURI(AUTHORITY, "my_downloads/#/" + DownloadContract.RequestHeaders.URI_SEGMENT, REQUEST_HEADERS_URI);
+        URI_MATCHER.addURI(AUTHORITY, "all_downloads/#/" + DownloadContract.RequestHeaders.URI_SEGMENT, REQUEST_HEADERS_URI);
+        // temporary, for backwards compatibility
+        URI_MATCHER.addURI(AUTHORITY, "download", MY_DOWNLOADS);
```

Comment:
seems like this and the ones below could be deleted since we don't need any backwards compatibility